File: src/main/java/com/everyonewaiter/domain/auth/entity/AuthSuccess.java
package com.everyonewaiter.domain.auth.entity;

import java.time.Duration;

public record AuthSuccess(
    String phoneNumber,
    AuthPurpose purpose,
    Duration expiration
) implements Auth {

  private static final String KEY_PREFIX = "auth:success:";

  public AuthSuccess(String phoneNumber, AuthPurpose purpose) {
    this(phoneNumber, purpose, purpose.getExpiration());
  }

  @Override
  public String getKey() {
    return KEY_PREFIX + phoneNumber;
  }

  @Override
  public int getValue() {
    return -2;
  }

}

File: src/main/kotlin/com/everyonewaiter/domain/auth/entity/AuthSuccess.kt
package com.everyonewaiter.domain.auth.entity

import java.time.Duration

data class AuthSuccess(
    val phoneNumber: String,
    val expiration: Duration = Duration.ofMinutes(15),
) {
    val key: String
        get() = "auth:success:$phoneNumber"

    val value: String
        get() = phoneNumber
}
